stable/traefik/templates/hpa.yaml

```diff
@@ -0,0 +1,20 @@
+{{- if .Values.autoscaling }}
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
```

Comment:
Does this require a kubernetes version check? How far back is this supported?

Reply:
Good point, this is indeed kubernetes version dependent, for example autoscaling/v2beta2 is only supported from 1.11 forward, although not used here. Will add a variable in handlers to use it here.

Reply:
@dtomcej, reading the documentation a little bit more, the `apiVersion` can have values from v1 to v2beta2 and the main change is the `metrics` key in the spec, if we are allowing to define the whole metrics value then we should allow to define what apiVersion is using. Maybe adding `autoscaling.apiVersion` or something like that, what do you think?

Reply:
@dtomcej, any input on my comment?